pyanaconda/startup_utils.py

```diff
@@ -661,16 +670,69 @@ def start_geolocation_conditionally(opts):
     return geoloc_task_proxy
 
 
-def wait_for_geolocation(geoloc_task_proxy):
-    """Wait for geolocation, if started.
+def wait_for_geolocation_and_use(geoloc_task_proxy, display_mode):
+    """Wait for geolocation and use the result, if started.
 
     :param geoloc_task_proxy: D-Bus proxy for a GeolocationTask instance
+    :param display_mode: a display mode to use for the check
     """
     if not geoloc_task_proxy:
         return
 
     try:
         wait_for_task(geoloc_task_proxy, timeout=constants.GEOLOC_CONNECTION_TIMEOUT)
-        # TODO: set locale here instead of welcome spoke, rename helper
     except TimeoutError:
         log.debug("Geolocation timed out. Exceptions will not be logged.")
+        return
+    else:
+        apply_geolocation_result(display_mode)
+
+
+def apply_geolocation_result(display_mode):
+    """Apply geolocation result.
+
+    This does not check for kickstart installations, because in that case geolocation is not even
```

Comment:
Ah, I see. The comment really is a bit misleading. The thing is, `apply_geolocation_result`(this) is one of the multiple parts involved, and the last one, so it does not get to choose much :-)

Here's the whole picture:
- First, you call `start_geolocation_conditionally` that handles the starting, and damage control in case modules are not available etc.
- That actually calls `check_if_geolocation_should_be_used` to determine if geolocation should be started, which is where the kickstart status and boot option is actually considered.
- Then, you have `wait_for_geolocation_and_use` that actually handles the task and its finalization, and calls `apply_geolocation_result`.
- Once you are finally in `apply_geolocation_result` which you're commenting here, you just apply the result, the decision to start or not was already taken earlier.

I will think about describing this better.